packages/hint/src/bin/hint.ts

```diff
@@ -13,37 +13,73 @@
  */
 
 const debug = (process.argv.includes('--debug'));
+const tracking = (/--tracking[=\s]+([^\s]*)/i).exec(process.argv.join(' '));
+const analyticsDebug = process.argv.includes('--analytics-debug');
 
 import * as d from 'debug';
 
+import * as insights from '../lib/utils/appinsights';
+
 // This initialization needs to be done *before* other requires in order to work.
 if (debug) {
     d.enable('hint:*');
 }
 
+const trackingEnv = process.env.HINT_TRACKING; // eslint-disable-line no-process-env
+let enableTracking;
+
+if (tracking) {
+    enableTracking = tracking[1] === 'on';
+} else if (trackingEnv) {
+    enableTracking = trackingEnv === 'on';
+}
+
+if (enableTracking !== void 0) {
```

Comment:
`void 0 === undefined`

Reply:
Why not use `undefined` (or `typeof enableTracking !== 'undefined'` if ESLint is complaining)?

Reply:
`void 0` is shorten than `typeof enableTracking !== 'undefined'` hehehe

I read somewhere sometime ago that `void 0` is the best way to do it.

Reply:
> void 0 is shorten than typeof enableTracking !== 'undefined' hehehe

Yes, but code that is more readable and understood by more people is better. 

The reason why I asked is for consistency. We are using `undefined` everywhere else.

> I read somewhere sometime ago that void 0 is the best way to do it.

Do you have a link? 

I know in the past `void 0` was used instead of `undefined` because `undefined` was mutable, but since ES5 that is no longer the case (see also: https://eslint.org/docs/rules/no-void).

Reply:
> Do you have a link?

Nop

> I know in the past void 0 was used instead of undefined because undefined was mutable, but since ES5 that is no longer the case (see also: https://eslint.org/docs/rules/no-void).

Then we should use `undefined` directly, instead of `typeof` or `void 0`

Reply:
> Then we should use undefined directly, instead of typeof or void 0

I said `typeof` because we [have](https://github.com/webhintio/hint/blob/da79178d8753755ed6d29d28f441d9ab49dc06e2/.eslintrc.json#L174): [`"no-undefined": "error"`](https://eslint.org/docs/rules/no-undefined) in our `.eslintrc.json`.

Reply:
maybe we should remove that rule.

What do you think @webhintio/core?

Reply:
> maybe we should remove that rule.

I'm fine either way, I don't have a preference.

Reply:
We have a few `typeof XXX === 'undefined'` in the code that we should then update. I'm fine either way but that should probably be done in a different PR.